lib/showoff.rb

```diff
@@ -437,85 +435,170 @@ def process_markdown(name, section, content, opts={:static=>false, :pdf=>false,
+            # Change setting and return empty string
+            ''
+          }
+        },
+        # Load and replace any file tags
+        "file" => {
+          "matchers" => [
+            /~~~FILE:(?<name>[^:~]*):?(?<languages>.*)?~~~/
+          ],
+          "block" => proc { |match_data|
+            # make a list of code highlighting classes to include
+            css = match_data[:languages].split.collect {|l| "language-#{l.downcase}" }.join(' ')
+
+            # get the file content and parse out html entities
+            file = File.read(File.join(settings.pres_dir, '_files', match_data[:name])) rescue "Nonexistent file: #{match_data[:name]}"
+            file = "Empty file: #{match_data[:name]}" if file.empty?
+            file = HTMLEntities.new.encode(file) rescue "HTML parsing of #{match_data[:name]} failed"
+
+            %Q(<pre class="highlight"><code class="#{css}">#{file}</code></pre>)
+          }
+        }
+      }
+    end
+
+
     # This method processes the content of the slide and replaces
     # content markers with their actual value information
     def process_content_for_replacements(content)
-      # update counters, incrementing section:minor if needed
-      result = content.gsub("~~~CURRENT_SLIDE~~~", @slide_count.to_s)
-      result.gsub!("~~~SECTION:MAJOR~~~", @section_major.to_s)
-      if result.include? "~~~SECTION:MINOR~~~"
-        @section_minor += 1
-        result.gsub!("~~~SECTION:MINOR~~~", @section_minor.to_s)
-      end
-
-      # scan for pagebreak tags. Should really only be used for handout notes or supplemental materials
-      result.gsub!("~~~PAGEBREAK~~~", '<div class="pagebreak">continued...</div>')
-
-      # replace with form rendering placeholder
-      result.gsub!(/~~~FORM:([^~]*)~~~/, '<div class="form wrapper" title="\1"></div>')
-
-      # Now check for any kind of options
-      content.scan(/(~~~CONFIG:(.*?)~~~)/).each do |match|
-        result.gsub!(match[0], settings.showoff_config[match[1]]) if settings.showoff_config.key?(match[1])
-      end
-
-      # Load and replace any file tags
-      content.scan(/(~~~FILE:([^:~]*):?(.*)?~~~)/).each do |match|
-        # make a list of code highlighting classes to include
-        css  = match[2].split.collect {|i| "language-#{i.downcase}" }.join(' ')
-
-        # get the file content and parse out html entities
-        name = match[1]
-        file = File.read(File.join(settings.pres_dir, '_files', name)) rescue "Nonexistent file: #{name}"
-        file = "Empty file: #{name}" if file.empty?
-        file = HTMLEntities.new.encode(file) rescue "HTML parsing of #{name} failed"
+      return unless content
 
-        result.gsub!(match[0], "<pre class=\"highlight\"><code class=\"#{css}\">#{file}</code></pre>")
+      result = content
+      custom_syntax_templates.each do |name, syntax|
+        syntax['matchers'].each do |matcher|
+          result.gsub!(matcher) {
+            syntax['block'].($~)
```

Comment:
Probably good to handle exceptions here, in case a proc or matcher or some such is defined incorrectly.